src/main/java/org/jenkinsci/plugins/envinject/EnvInjectEnvVarsContributor.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Extension
+public class EnvInjectEnvVarsContributor extends EnvironmentContributor {
+
+    @Override
+    public void buildEnvironmentFor(Job job, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        for (Object property : job.getProperties().values()) {
```

Comment:
@oleg-nenashev AFAIK, if you are using the plugin (checkbox is clicked), there is always a `EnvInjectJobProperty`. And I've verified that `job.getProperties()` can never be `null`.